### PR TITLE
Add support for js-codegen in fuzzer

### PR DESCRIPTION
**Pull request by jarivm**
Integrate js-codegen in fuzzer. A similar approach to choosing a language for a `FuzzCase` is chosen, but instead of a language, a backend is randomly picked.

### CODE REVIEW

**Comment by DrRataplan** [comment]
## BundleMon

  
  
  
  
  
  
  
<details>
<summary>Unchanged files (2)</summary>

Status | Path | Size | Limits
:------------: | ------------ | :------------: | :------------:
:white_check_mark: | fontoxpath.esm.js<br/> | 91.13KB | -
:white_check_mark: | fontoxpath.js<br/> | 91.02KB | -
</details>



No change in files bundle size

Final result: :white_check_mark:

---
<p align="center"><a href="https://bundlemon.now.sh/projects/609024effb87a20009a6b096/reports?branch=support-js-codegen-in-fuzzer&resolution=all" target="_blank" rel="noreferrer noopener">Current branch size history</a> | <a href="https://bundlemon.now.sh/projects/609024effb87a20009a6b096/reports?branch=master&resolution=days" target="_blank" rel="noreferrer noopener">Target branch size history</a></p>

File: fuzzers/mutators.ts [hunk 1]
```diff
@@ -29,6 +29,26 @@ export function randomLanguage(): string {
 	}
 }
 
+// We use string values because they can be printed when reporting errors.
+export type Backend = 'expression' | 'js-codegen';
```
Comment (devatwork) [inline fuzzers/mutators.ts:33]: Why u no enum?

**Review by devatwork — APPROVED** [review]
LGTM. One minor thing.

**Comment by devatwork** [comment]
Thanks for the PR @jarivm! Did the fuzzer find anything in your code?

File: fuzzers/mutators.ts (hunk 1 above)
Comment (jarivm) [reply fuzzers/mutators.ts:33]: I assumed enums could only have number values. Just looked it up and I learned I can use strings too. I'm going to change it.

**Comment by jarivm** [comment]
> Thanks for the PR @jarivm! Did the fuzzer find anything in your code?

No, actually! The fuzzer only found errors already reported by you.

**Review by DrRataplan — APPROVED** [review]
LGTM

**Pull request merged**